JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -10,91 +10,310 @@
+import java.net.URLEncoder;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.vertica.jdbc.VerticaConnection;
+
+public class OAuthSampleApp {
+	private static Properties prop;
+	private static Map<String, String> csProp = new HashMap<String, String>();
+	private static String OAUTH_ACCESS_TOKEN_VAR_STRING = "OAUTH_SAMPLE_ACCESS_TOKEN";
+	private static String OAUTH_REFRESH_TOKEN_VAR_STRING = "OAUTH_SAMPLE_REFRESH_TOKEN";
+	private static Connection vConnection;
+
+	// Get jdbc connection string from provided configuration
+	private static void getcsProp(String args) {
+		String[] entries = args.split(";");
+		for (String entry : entries) {
+			String[] pair = entry.split("=");
+			csProp.put(pair[0], pair[1]);
+		}
+	}
+
+	private static String getConnectionString() {
+		return "jdbc:vertica://" + csProp.get("Host") + ":" + csProp.get("Port") + "/" + csProp.get("Database")
+				+ "?user=" + csProp.get("User") + "&password=" + csProp.get("Password");
+	}
+
+	// Get the parameters from the connection String
+	private static String getParam(String paramName) {
+		return csProp.get(paramName);
+	}
+
+	// Add/Create Authentication record in database. Create User and grant the
+	// permissions for User
+	private static void SetUpDbForOAuth() throws Exception {
+		String connectionString = getConnectionString();
+		vConnection = DriverManager.getConnection(connectionString);
+		String USER = prop.getProperty("User");
+		String CLIENT_ID = prop.getProperty("ClientId");
+		String CLIENT_SECRET = prop.getProperty("ClientSecret");
+		String DISCOVERY_URL = prop.getProperty("DiscoveryUrl");
+		Statement st = vConnection.createStatement();
+		st.execute("DROP USER IF EXISTS " + USER + " CASCADE;");
+		st.execute("DROP AUTHENTICATION IF EXISTS v_oauth CASCADE;");
+		st.execute("CREATE AUTHENTICATION v_oauth METHOD 'oauth' LOCAL;");
+		st.execute("ALTER AUTHENTICATION v_oauth SET client_id= '" + CLIENT_ID + "';");
+		st.execute("ALTER AUTHENTICATION v_oauth SET client_secret= '" + CLIENT_SECRET + "';");
+		st.execute("ALTER AUTHENTICATION v_oauth SET discovery_url = '" + DISCOVERY_URL + "';");
+		st.execute("CREATE USER " + USER + ";");
+		st.execute("GRANT AUTHENTICATION v_oauth TO " + USER + ";");
+		st.execute("GRANT ALL ON SCHEMA PUBLIC TO " + USER + ";");
+		st.close();
+	}
+
+	// Dispose the authentication record from database
+	private static void TearDown() {
+		try {
+			Statement st = vConnection.createStatement();
+			String USER = prop.getProperty("User");
+			st.executeUpdate("DROP USER IF EXISTS " + USER + " CASCADE");
+			st.executeUpdate("DROP AUTHENTICATION IF EXISTS v_oauth CASCADE");
+			vConnection.close();
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	// Connect to Database using access Token
+	private static Connection connectToDB(String accessToken) throws SQLException {
```

Comment:
You have a function called "connectToDB" and one called "ConnectToDatabase". This would be confusing for users.

Reply:
Changed function name from connecToDb to getConnection.